Helper/Data.php

```diff
@@ -108,56 +56,35 @@ public function getApiSecret($storeId)
         );
     }
 
-    /**
-     * Get API Secret from system configuration
-     *
-     * @return string
-     */
     public function getApiEndpoint()
     {
         return $this->config->getValue(
             'metrilo_analytics/general/api_endpoint'
         );
     }
 
-    /**
-     * Get session data with "metrilo_events" key
-     *
-     * @return array
-     */
-    public function getSessionEvents()
+    public function getSessionEvents($type)
```

Comment:
I find this session-related logic confusing and I don't think everything should be in the same place.
What do you think about adding a helper that has the responsibility of dealing with sessions?

Reply:
Added new cartEvents helper responsible for instantiating catalog session and add/get metrilo cart cations.